### PR TITLE
store user selected account types in ipfs library sc

### PR DESCRIPTION
**Pull request by sargonpiraev**
we need to store user selected account types
because we need to fetch correct worker jobs, client jobs.
now its implemented by faker.js

after registration we need to store somewhere user selected account types - worker, client, recruiter. now we can implement this feature by IPFSLibrary sc.
create dao for ipfs library sc
on registration create ipfs hash and store it in ipfs library sc
on login get account types from ipfs library, than from ipfs
fix separate job loading of worker and client
fix second menu. show only allowed menu items

### CODE REVIEW

File: src/daos/lib/IPFSLibraryDAO.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      this.hashSetEmitter = null
+    }
+  }
+
+  get isConnected () {
+    return this.contract != null // nil check
+  }
+
+  async getHash (from, itemName) {
+    const itemNameInBytes32 = web3.utils.asciiToHex(itemName)
```
Comment (ipavlenko) [inline src/daos/lib/IPFSLibraryDAO.js]: https://github.com/ChronoBank/Labor-X/blob/develop/src/utils/bytes32.js#L7
Does this action make exactly the same thing?

File: src/models/app/HashSetEvent.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  self: PropTypes.string.isRequired,
+  from: PropTypes.string.isRequired,
+  itemName: PropTypes.string.isRequired,
+  itemHash: PropTypes.string.isRequired,
+})
+
+export default class HashSetEvent extends AbstractModel {
+  constructor (props) {
+    super(props, schemaFactory())
+    Object.assign(this, props)
```
Comment (ipavlenko) [inline src/models/app/HashSetEvent.js]: odd string

File: src/store/createAccount/actions.js [hunk 3]
```diff
@@ -24,23 +24,23 @@ export const setAccountTypes = (types) => (dispatch) => {
 
 export const generateNameWallet = () => (dispatch, getState) => {
   const state = getState()
-  
+
   const { walletsList } = state.wallet
-  
+
   let currentName
```
Comment (ipavlenko) [inline src/store/createAccount/actions.js]: Create freeNameSelector, move implementation there

File: src/store/login/actions.js [hunk 4]
```diff
@@ -61,12 +61,12 @@ export const signOut = () => (dispatch) => {
...
 }
 
-export const createAccount = (walletName, password) => (dispatch, getState) => {
+export const createAccount = (walletName, password) => async (dispatch, getState) => {
   const state = getState()
   const signInModel = state.login.signIn
 
   if (signInModel) {
-    const wallet = dispatch(createWallet({ [signInModel.method] : signInModel.key,  name: walletName, password: password }))
+    const wallet = await dispatch(createWallet({ [signInModel.method] : signInModel.key,  name: walletName, password: password }))
```
Comment (ipavlenko) [inline src/store/login/actions.js]: Please split into several lines

File: src/store/user/actions.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+export const USER_SAVE = 'user/save'
+
+export const userSave = (user) => ({ type: USER_SAVE, user })
+
+export const getUserData = (address) => async (dispatch, getState) => {
+
+  const state = getState()
+
+  const IPFSLibrary = daoByType('IPFSLibrary')(state)
```
Comment (ipavlenko) [inline src/store/user/actions.js]: prefer camelCase naming for variables

File: src/store/user/reducer.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+export const initialState = null
+
+const persistConfig = () => ({
+  key: 'user',
+  storage: storage,
+})
+
+const mutations = {
+  [USER_SAVE] (state, { user }) {
+    return { ...user }
```
Comment (ipavlenko) [inline src/store/user/reducer.js]: Make the Immutable (use Object.freeze(...)) model for User.
Do not expand user.
You must not expand immutable objects before persisting to redux state

File: src/daos/lib/IPFSLibraryDAO.js (hunk 1 above)
Comment (sargonpiraev) [reply src/daos/lib/IPFSLibraryDAO.js]: imho these algorythms differs
https://github.com/cryptocoinjs/base-x/blob/master/index.js#L51
https://github.com/ethereum/web3.js/blob/v1.0.0-beta.33/packages/web3-utils/src/index.js

found here instructions how to convert string to bytes32 and back 
https://ethereum.stackexchange.com/questions/23058/web3-return-bytes32-string

File: src/models/app/HashSetEvent.js (hunk 2 above)
Comment (sargonpiraev) [reply src/models/app/HashSetEvent.js]: +

File: src/store/login/actions.js (hunk 4 above)
Comment (sargonpiraev) [reply src/store/login/actions.js]: fixed

File: src/store/user/reducer.js (hunk 6 above)
Comment (sargonpiraev) [reply src/store/user/reducer.js]: fixed

File: src/store/user/actions.js (hunk 5 above)
Comment (sargonpiraev) [reply src/store/user/actions.js]: fixed

File: src/daos/lib/IPFSLibraryDAO.js (hunk 1 above)
Comment (ipavlenko) [reply src/daos/lib/IPFSLibraryDAO.js]: Everything fine

File: src/store/createAccount/actions.js (hunk 3 above)
Comment (sargonpiraev) [reply src/store/createAccount/actions.js]: fixed

**Pull request merged**